banyan/resource_policy.go

```diff
@@ -0,0 +1,500 @@
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"template": {
+							Type:         schema.TypeString,
+							Optional:     true,
+							ValidateFunc: validatePolicyTemplate(),
+						},
+					},
+				},
+			},
+			"spec": {
+				Type:        schema.TypeList,
+				MinItems:    1,
+				MaxItems:    1,
+				Required:    true,
+				Description: "The spec",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"access": {
+							Type:        schema.TypeList,
+							MinItems:    0,
+							Optional:    true,
+							Description: "access",
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"roles": {
+										Type: schema.TypeSet,
+										Elem: &schema.Schema{
+											Type: schema.TypeString,
+										},
+										Optional: true,
+									},
+									"rules": {
+										Type:        schema.TypeList,
+										MinItems:    1,
+										MaxItems:    1,
+										Optional:    true,
+										Description: "rules for enforcing security policy",
+										Elem: &schema.Resource{
+											Schema: map[string]*schema.Schema{
+												"conditions": {
+													Type:     schema.TypeList,
+													MinItems: 1,
+													MaxItems: 1,
+													Required: true,
+													Elem: &schema.Resource{
+														Schema: map[string]*schema.Schema{
+															"trust_level": {
+																Type:         schema.TypeString,
+																Required:     true,
+																ValidateFunc: validateTrustLevel(),
+															},
+														},
+													},
+												},
+												"l7_access": {
+													Type:        schema.TypeList,
+													MinItems:    1,
+													Optional:    true,
+													Description: "ha",
+													Elem: &schema.Resource{
+														Schema: map[string]*schema.Schema{
+															"resources": {
+																Type:     schema.TypeSet,
+																Required: true,
+																Elem: &schema.Schema{
+																	Type: schema.TypeString,
+																},
+															},
+															"actions": {
+																Type:     schema.TypeSet,
+																Required: true,
+																Elem: &schema.Schema{
+																	Type: schema.TypeString,
+																},
+																//TODO validate actions? ValidateFunc: validatePort(),
+															},
+														},
+													},
+												},
+											},
+										},
+									},
+								},
+							},
+						},
+						"exception": {
+							Type:        schema.TypeList,
+							MaxItems:    1,
+							MinItems:    1,
+							Required:    true,
+							Description: "HTTP settings used for x",
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"src_addr": {
+										Type:     schema.TypeSet,
+										Optional: true,
+										Elem: &schema.Schema{
+											Type: schema.TypeString,
+										},
```

Comment:
validate each elem with both regexes

```
regexp.MustCompile(`^(http:\/\/www\.|https:\/\/www\.|http:\/\/|https:\/\/)?[a-z0-9]+([\-\.]{1}[a-z0-9]+)*\.[a-

regexp.MustCompile(`^[0-9\/\.]+$`)
```

Reply:
cant actually do this here. so we validate before sending over the wire
see for what is actually happening https://github.com/hashicorp/terraform-plugin-sdk/issues/156